src/clientSideScene/sceneEntities.ts

```diff
@@ -1444,9 +1445,11 @@ export class SceneEntities {
         await kclManager.executeAstMock(astSnapshot)
         await codeManager.updateEditorWithAstAndWriteToFile(astSnapshot)
 
-        sceneInfra.modelingSend({ type: 'circle3PointsFinished', cleanup })
+        done()
```

Comment:
My only concern is, what if there's some kind of error in this and `done()` never gets called?

Reply:
Will address in a follow up; completely right :)